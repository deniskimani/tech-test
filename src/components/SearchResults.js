/* eslint-disable jsx-a11y/img-redundant-alt */
import "../styles/search-results.css";
const SearchResults = ({ results, input }) => {
  if (!results.length) {
    if (!input) {
      return <p className="unavailable">Please type something in search box</p>;
    } else {
      return (
        <p className="unavailable">
          No results for <span>{input}</span>!
        </p>
      );
    }
  } else {
    return (
      <div className="container">
        <p className="available">
          Search results for <span>{input}</span>
        </p>
        {results.map((image) => (
          <img className="card-image" src={image} alt="spaceImage" />
        ))}
      </div>
    );
  }
};

export default SearchResults;
