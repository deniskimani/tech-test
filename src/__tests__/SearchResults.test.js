import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search", () => {
  const validProps = [
    "https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg",
  ];
  const { asFragment } = render(<SearchResults results={validProps} />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
