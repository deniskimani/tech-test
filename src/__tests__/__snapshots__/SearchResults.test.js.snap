// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="spaceImage"
      class="card-image"
      src="https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg"
    />
    <img
      alt="spaceImage"
      class="card-image"
      src="https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg"
    />
  </div>
</DocumentFragment>
`;
